modules/imgproc/src/imgwarp.cpp

```diff
@@ -441,190 +441,167 @@ struct RemapVec_8u
     {
         int cn = _src.channels(), x = 0, sstep = (int)_src.step;
 
-        if( (cn != 1 && cn != 3 && cn != 4) || !checkHardwareSupport(CV_CPU_SSE2) ||
+        if( (cn != 1 && cn != 3 && cn != 4) || !hasSIMD128() ||
             sstep > 0x8000 )
             return 0;
 
         const uchar *S0 = _src.ptr(), *S1 = _src.ptr(1);
         const short* wtab = cn == 1 ? (const short*)_wtab : &BilinearTab_iC4[0][0][0];
         uchar* D = (uchar*)_dst;
-        __m128i delta = _mm_set1_epi32(INTER_REMAP_COEF_SCALE/2);
-        __m128i xy2ofs = _mm_set1_epi32(cn + (sstep << 16));
-        __m128i z = _mm_setzero_si128();
+        v_int32x4 delta = v_setall_s32(INTER_REMAP_COEF_SCALE / 2);
+        v_int16x8 xy2ofs = v_reinterpret_as_s16(v_setall_s32(cn + (sstep << 16)));
         int CV_DECL_ALIGNED(16) iofs0[4], iofs1[4];
 
+#define CV_PICK_AND_PACK_RGB(ptr, offset, result) \
+        {                                             \
+            unsigned char* p = (unsigned char*)ptr;   \
+            result = v_int16x8((short)p[offset],   (short)p[offset+3], /* r0r1 */ \
+                               (short)p[offset+1], (short)p[offset+4], /* g0g1 */ \
+                               (short)p[offset+2], (short)p[offset+5], /* b0b1 */ 0, 0); \
```

Comment:
Buffer overrun is very rare case here, so we can use check here:
```
#define CV_PICK_AND_PACK_RGB(ptr, offset, result) \
        { \
            const uchar* const p = ((const uchar*)ptr) + (offset); \
            if (p <= src_limit_8bytes) \
            { \
                v_uint8x16 rrggbb, dummy; \
                v_uint16x8 rrggbb8, dummy8; \
                v_uint8x16 rgb0 = v_reinterpret_as_u8(v_int32x4(*(int*)(p), 0, 0, 0)); \
                v_uint8x16 rgb1 = v_reinterpret_as_u8(v_int32x4(*(int*)(p + 3), 0, 0, 0)); \
                v_zip(rgb0, rgb1, rrggbb, dummy); \
                v_expand(rrggbb, rrggbb8, dummy8); \
                result = v_reinterpret_as_s16(rrggbb8); \
            } \
            else \
            { \
                result = v_int16x8((short)p[0], (short)p[3], /* r0r1 */ \
                                   (short)p[1], (short)p[4], /* g0g1 */ \
                                   (short)p[2], (short)p[5], /* b0b1 */ 0, 0); \
            } \
        }

where

const uchar* src_limit_8bytes = _src.datalimit - 8;
```

Alternatively we can try reading "int" + "short casted to int".

Reply:
Thank you, I followed your comment